docs/source/tutorial_fast.md

```diff
@@ -0,0 +1,150 @@
+        --embedding-size 250 --ffn-size 1000 --n-layers 8 -opt adam -lr 1e-3 \
+        --skip-generation true \
+        --eval-batchsize 128 \
+        --dynamic-batching true --truncate 256
+
+You should notice that your memory utilization is much higher in this mode.
+This actually has an advantage, since you can more easily find your maximum
+batchsize.  To get the full benefits of dynamic batching, make sure to use the
+largest batchsize you can.
+
+Overall, this results in a large increase in speed: about 2x, bringing training
+down to 250s and evaluation to 11s.
+
+_WARNING_: You may find perplexity is quite a bit worse than without dynamic
+batching. This is because we use larger batches, and take fewer steps.  You can
+usually increase your learning rate pretty substantially when using dynamic
+batching, to compensate for the fewer steps.
+
+## FP16
+
+If you have access to an NVIDIA GPU with FP16 CUDA Cores (V100, GTX 2080, etc),
+then you can get large speedups by switching on the option `--fp16 true`. The
+default version of FP16 requires that you install
+[APEX](https://github.com/NVIDIA/apex), but you can use a simplified version
+(which doesn't depend on APEX) with `--fp16 true --fp16-impl mem_efficienet`.
+
+Note that in order to get the full benefit of fp16, we need to make sure all
+our hidden dimensions are _multiples of 8_, otherwise the hardware won't use
+CUDA cores. We will slightly adjust the size of the network to support this.
+We'll slightly adjust the network parameters (--embedding-size and --ffn-size)
+to conform to this.
+
+    parlai train -df dictfile -m transformer/generator -t convai2 -eps 1.0 -bs 64 \
+        --embedding-size 256 --ffn-size 1024 --n-layers 8 -opt adam -lr 1e-3 \
+        --skip-generation true \
+        --eval-batchsize 128 \
+        --dynamic-batching true \
+        --fp16 true --fp16-impl mem_efficient
+
+Further notice that FP16 often significantly lowers the memory size of your model
+and activations (almost by a factor of 2). This means you can usually get away with
+significantly increasing the batchsize (and eval batchsize).
+
+    parlai train -df dictfile -m transformer/generator -t convai2 -eps 1.0 \
+        --embedding-size 256 --ffn-size 1024 --n-layers 8 -opt adam -lr 1e-3 \
+        --skip-generation true \
+        --eval-batchsize 256 \
+        --dynamic-batching true \
+        --fp16 true -bs 128
+
+In this example, we see about a 25% speedup. Generally you can expect a larger
+speedup with larger models, with models of >300M often getting a ~50% speedup.
+With the increased batch size, this can often be brought to 2.5x faster.
+
+## Use multiple GPUs
+
+If you have multiple GPUs, you can utilize them by switching from `train` to
+`multiprocessing_train`. If you have 4 GPUs, you'll find training should be
+roughly 3.5x faster. The arguments for the training are left otherwise the same.
+
+    parlai multiprocessing_train \
```

Comment:
This does not work as well due to the naming of model multiprocess_train,  calling this 
 https://github.com/facebookresearch/ParlAI/blob/master/parlai/utils/strings.py#L67 will produce the error message:  ``` module 'parlai.scripts.multiprocessing_train' has no attribute 'MultiprocessingTrain' ``` whereas the actual class name is ``` MultiProcessingTrain ```. I would suggest we rename multiprocessing_train as multi_processing_train

Reply:
t

Reply:
It works after #2772 lands.